telemetry_package/main.py

```diff
@@ -1,31 +1,140 @@
+
+
+class MinimalPublisher(Node):
 
-from common_package_py.common_node import CommonNode
+    def __init__(self):
+        super().__init__('telemetry_node')
+        self.subscription = self.create_subscription(
+            Log,
+            '/rosout',
+            self.run_server, 
+            100)
+        self.subscription 
 
+        # Counter for debugging purposes
+        self.msg_cnt = 0
 
-class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
+        self.msg_cache = list()
+        self.cache_size = 400
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
 
-        Args:
-            id (str): Unique node id
+    def is_socket_closed(self, sock: socket.socket) -> bool:
+        try:
+            # this will try to read bytes without blocking and also without removing them from buffer (peek only)
+            data = sock.recv(16, socket.MSG_DONTWAIT | socket.MSG_PEEK)
+            if len(data) == 0:
+                return True
+        except BlockingIOError:
+            return False  # socket is open and reading from it would block
+        except ConnectionResetError:
+            return True  # socket was closed for some other reason
+        except Exception as e:
+            self.get_logger().error(f"Unexpected exception when checking if the socket is closed: {e}")
+            return False
+        return False
+    
+
+    '''
+    Code to implement the websocket functionality     
+    '''
+    def socket_exists(self, socket_path: str) -> bool:
+        return os.path.exists(socket_path)
+
+
+    def send_message(self, client_socket):
         """
+        Defines the message block to be continuosly send.
+        """
+        if len(self.msg_cache) > 0:
+            client_socket.sendall(json.dumps(self.msg_cache[0]).encode())
+            self.get_logger().info(f'Send message #{self.msg_cnt + 1}')
+            self.msg_cnt += 1     
+            self.msg_cache.pop(0)  
+
+
+    def run_server(self, msg, socket_path=DEFAULT_UNIX_SOCKET_PATH):
+        # Check if socket is in use
+        if self.socket_exists(socket_path):
+            self.get_logger().fatal(f'Socket path already exists')
+            raise FileExistsError('')
+
+        # Create a socket
+        server_socket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+
+        try:
+            # Bind socket
+            server_socket.bind(socket_path)
+            self.get_logger().info(f'Binding socket.')
+
+            # Listen for connection
+            server_socket.listen(1)
+            self.get_logger().info(f'Listening for connection on {socket_path} "...')
+
+            while True:
+                # Appending the stream of sent messages so that the socket can take that amount of traffic
+                self.msg_cache.append(msg.data)
+                self.get_logger().info(f'Received message with following payload: {self.msg_cache[len(self.msg_cache) - 1]}')
```

Comment:
If you want to access the last element of a list, just use [-1]
```suggestion
                self.get_logger().info(f'Received message with following payload: {self.msg_cache[-1]}')
```